clouddriver-web/src/test/groovy/com/netflix/spinnaker/clouddriver/requestqueue/pooled/PooledRequestQueueSpec.groovy

```diff
@@ -35,12 +37,51 @@ class PooledRequestQueueSpec extends Specification {
 
   def "should time out if request does not complete"() {
     given:
-    def queue = new PooledRequestQueue(new NoopRegistry(), 10, 1)
+    def queue = new PooledRequestQueue(new NoopRegistry(), 5000, 10, 1)
 
     when:
-    Long result = queue.execute("foo", { Thread.sleep(20); return 12345L })
+    queue.execute("foo", { Thread.sleep(20); return 12345L })
 
     then:
     thrown(PromiseTimeoutException)
   }
+
+
+  def "should time out if request does not start in time"() {
```

Comment:
its a 40 millisecond sleep thread that the test ensures has actually started before exercising the specific test case so I don't think there is a scenario where it ends up hanging around ever